deepchem/models/tensorgraph/models/symmetry_function_regression.py

```diff
@@ -122,17 +123,82 @@ def __init__(self,
     """
     self.n_tasks = n_tasks
     self.max_atoms = max_atoms
-    self.n_feat = n_feat
     self.layer_structures = layer_structures
     self.atom_number_cases = atom_number_cases
     super(ANIRegression, self).__init__(**kwargs)
     self.build_graph()
+    self.grad = None
+
+  def save(self):
+    self.grad = None # recompute grad on restore
+    super(ANIRegression, self).save()
+
+  def build_grad(self):
+    self.grad = tf.gradients(self.outputs, self.atom_feats)
+
+  def compute_grad(self, dataset, batch_size=1):
+    with self._get_tf("Graph").as_default():
+      if not self.built:
+        self.build()
+      if not self.grad:
+        self.build_grad()
+
+      feed_dict = dict()
+      X = dataset.X
+      flags = np.sign(np.array(X[:batch_size, :, 0]))
+      feed_dict[self.atom_flags] = np.stack([flags]*self.max_atoms, axis=2)*\
+          np.stack([flags]*self.max_atoms, axis=1)
+      feed_dict[self.atom_numbers] = np.array(X[:batch_size, :, 0], dtype=int)
+      feed_dict[self.atom_feats] = np.array(X[:batch_size, :, :], dtype=float)
+      return self.session.run([self.grad], feed_dict=feed_dict)
+
+  def pred_one(self, X, atomic_nums):
```

Comment:
Could you add a docstring here? (Since this isn't a method from the superclass that's overridden)

Reply:
i can certainly add a docstring for this as is, but I wasn't sure if this was the best API to expose vs something like pred_batch where multiple predictions can be made. Part of the reason it looks like this has to do with integration with scipy.optimize.minimize so we can pass in extra kwargs.

Reply:
I'm OK with either option; we can always fix up the API in a future PR as we gain better understanding of the correct structure.